actions/delete_node_handler.go

```diff
@@ -52,12 +61,48 @@ func (h *deleteNodeHandler) Handle(ctx context.Context, action *castai.ClusterAc
 	log.Info("deleting kubernetes node")
 
 	b := backoff.WithContext(backoff.WithMaxRetries(backoff.NewConstantBackOff(h.cfg.deleteRetryWait), h.cfg.deleteRetries), ctx)
-	return backoff.Retry(func() error {
+	err := backoff.Retry(func() error {
 		err := h.clientset.CoreV1().Nodes().Delete(ctx, req.NodeName, metav1.DeleteOptions{})
 		if apierrors.IsNotFound(err) {
 			log.Info("node not found, skipping delete")
 			return nil
 		}
 		return err
 	}, b)
+
+	if err != nil {
+		return err
+	}
+
+	pods, err := h.clientset.CoreV1().Pods(metav1.NamespaceAll).List(ctx, metav1.ListOptions{
+		FieldSelector: fields.SelectorFromSet(fields.Set{"spec.nodeName": req.NodeName}).String(),
+	})
+	if err != nil {
+		return fmt.Errorf("listing node pods %w", err)
+	}
+
+	log.Infof("node has %d pods - removing", len(pods.Items))
+
+	deleteOptions := metav1.NewDeleteOptions(0)
```

Comment:
What does '0' mean in these delete opts?

Reply:
grace period 0

Reply:
why not init a var to name that? Easier to read

Reply:
Added comment is enough or should I rename var?